lib/benchmarks/exe/restore-bench.hs

```diff
@@ -1039,42 +1061,43 @@ reportProgress nw tr targetSync readSlot =  do
     posixNow <- utcTimeToPOSIXSeconds <$> getCurrentTime
     progress <- readSlot >>= syncProgress nw
     traceWith tr $ MsgRestorationTick posixNow progress
-    threadDelay 1000000
+    threadDelay 1_000_000
     if progress > Syncing (Quantity targetSync)
         then return ()
         else reportProgress nw tr targetSync readSlot
 
 -- | Poll the network tip until it reaches the slot corresponding to the current
 -- time.
 waitForNodeSync
-    :: forall n. Tracer IO (BenchmarkLog n)
+    :: ToTipTimeout
+    -> Tracer IO (BenchmarkLog n)
     -> NetworkLayer IO Block
     -> IO SlotNo
-waitForNodeSync tr nw = loop 960 -- allow 240 minutes for first tip
+waitForNodeSync ttr tr nw = loop $ maybeTipTimeout ttr 1 id
   where
     loop :: Int -> IO SlotNo
-    loop retries = do
+    loop timeout = do
         nodeTip <- currentNodeTip nw
         case nodeTip of
             Read.GenesisTip ->
-                if retries > 0
-                then do
-                    let delay = 15000000
-                    traceWith tr $ MsgRetryShortly delay
-                    threadDelay delay
-                    loop (retries - 1)
-                else
-                    throwString "Gave up in waitForNodeSync, waiting a tip"
+                if timeout > 0
+                    then do
+                        maybeTipTimeout ttr (loop timeout) $ \_ -> do
+                            let delay = 15_000_000
+                            traceWith tr $ MsgRetryShortly delay
+                            threadDelay delay
+                            loop (timeout - delay)
```

Comment:
🤨 According to the documentation of the option, `Nothing` should correspond to waiting indefinitely. But I don't quite follow this code. Could you change the type of `loop :: Maybe Int -> IO SlotNo` and handle the case directly, so that it becomes more clear?